src/lib/autocomplete/BUILD.bazel

```diff
@@ -72,3 +72,8 @@ markdown_to_html(
   name = "overview",
   srcs = [":autocomplete.md"],
 )
+
+filegroup(
+  name = "source-files",
+  srcs = glob(["**/*.ts"]),
+)
```

Comment:
The more I think about this, the more it seems like bazel is overkill for the docs generation. We end up with:
* 35 `filegroup` rules to expose the sources for syntax highlighting
* 54 `markdown_to_html` rules for the individual targets
* The starlark code to marshal arguments to the node implementation

The benefits of bazel also don't seem like they apply, since nothing really _depends_ on the docs. The caching isn't a big plus because the docs run relatively quickly as is (~8 seconds on my machine). 

Do you think that this switch is providing value that's worth it compared to everything that has to be setup up all around the repo? My thinking is that just a plain node script (with `npm run`) would be simpler.

Reply:
Hmm, we already chatted about it multiple times, and I don't even feel myself too strong about using Bazel for it (especially after needing `filegroup's` to expose the TS sources).

1) Bazel would definitely be able to cache the builds (e.g. when packaging the docs content; when transforming the markdown files; when building the API docs), but it's definitely **also true** that just running it without Bazel, is still fast enough and that's not a problem. Additionally we don't need caching because we don't build it frequently when developing.

2) I agree that it feels unnecessary to always pass the inputs/outputs to the NodeJS implementation using Starlark. This feels just unnecessary compared to how simple it is to just have a non-Bazel NodeJS script

3) I don't specifically consider the `markdown_to_html` rule as problematic. They pretty much fit into the whole bazel concept where they create an output from an input, and they can be built concurrently. Though it might not be necessary, but it just would be part of the "full" Bazel build.

4) My biggest point of doing it with Bazel was just to eventually do everything with Bazel. Since we eventually want to release with Bazel, it just feels right to also package the docs content into the Bazel `npm_package`. Just having it here as a target that builds an _uncomplete_ target feels wrong and I've been following the **Skydoc** concept kind of here. They could technically also generate docs with a much simpler setup I guess.

5) Technically the complete goal of a Bazel project would that _you only require_ Bazel as dependency and that everything will be managed by Bazel.. so requiring people to install NodeJS in order to build parts of the project, would go against that idea. Realistically, everybody has NodeJS installed though 😄 

I don't have too strong points here.. I personally like the way we would:

* Transform the `markdown` files to HTML (that's pretty reasonable to me personally)
* Highlight the files (this also fits good into the whole Bazel concept IMO)

but my **biggest** concern is still that we need so many filegroup's to expose the TS sources.

Reply:
As we discussed per chat. Since we both don't really have too strong points and both approaches have benefits, we are just keeping the existing work and going with the Bazel way for now.

For later reference: The reason I want to go this way now, rather than just moving to a plain NodeJS script is that the goal of the `src/material-examples:npm_package` Bazel target is to create a NPM package of it. If it would miss the docs content which needs to be generated as a separate build step, that goal is not achieved and we end up cobbling together.